sections/errorhandling/operationalvsprogrammererror.md

```diff
@@ -21,10 +23,37 @@ class AppError {
     this.isOperational = isOperational;
   }
 };
+<br/>
+throw new AppError(errorManagement.commonErrors.InvalidInput, "Describe here what happened", true);
 
+</pre>
+</details>
+
+<details>
+<summary><strong>Typescript</strong></summary>
+<pre lang="typescript">
+// some centralized error factory (see other examples at the bullet "Use only the built-in Error object")
+export class AppError extends Error {
+  public readonly commonType: string;
+  public readonly isOperational: boolean;
+<br/>
+  constructor(commonType: string, description: string, isOperational: boolean) {
+    super(description);
+<br/>
+    Object.setPrototypeOf(this, new.target.prototype); // restore prototype chain

```

Comment:
There is an explanation for this line in `useonlythebuiltinerror.md`. Should we state it again here?